etstool.py

```diff
@@ -0,0 +1,334 @@
+@click.option(
+    '--environment', default=None, help='Override the default environment name')
+def cleanup(runtime, toolkit, environment):
+    """ Remove a development environment.
+
+    """
+    parameters = get_parameters(runtime, toolkit, environment)
+    commands = [
+        "edm run -e {environment} -- python setup.py clean",
+        "edm environments remove {environment} --purge -y"]
+    click.echo("Cleaning up environment '{environment}'".format(**parameters))
+    execute(commands, parameters)
+    click.echo('Done cleanup')
+
+
+@cli.command()
+@click.option('--runtime', default='2.7', help='The python runtime version')
+@click.option(
+    '--toolkit', type=click.Choice(['pyside', 'pyqt4']),
+    default='pyside', help='The gui toolkit to use')
+@click.option(
+    '--environment', default=None, help='Override the default environment name')
+def test_clean(runtime, toolkit):
+    """ Run tests in a clean environment, cleaning up afterwards
+
+    """
+    args = ['--toolkit={}'.format(toolkit), '--runtime={}'.format(runtime)]
+    try:
+        install(args=args, standalone_mode=False)
+        test(args=args, standalone_mode=False)
+    finally:
+        cleanup(args=args, standalone_mode=False)
+
+
+@cli.command()
+@click.option('--runtime', default='2.7', help='The python runtime version')
+@click.option(
+    '--toolkit', type=click.Choice(['pyside', 'pyqt4']),
+    default='pyside', help='The gui toolkit to use')
+@click.option(
+    '--environment', default=None, help='Override the default environment name')
+def update(runtime, toolkit, environment):
+    """ Update/Reinstall package into environment.
+
+    """
+    parameters = get_parameters(runtime, toolkit, environment)
+    commands = [
+        "edm run -e {environment} -- python setup.py install"]
+    click.echo("Re-installing in  '{environment}'".format(**parameters))
+    execute(commands, parameters)
+    click.echo('Done update')
+
+
+# ----------------------------------------------------------------------------
+# Utility routines
+# ----------------------------------------------------------------------------
+
+
+def get_parameters(runtime, toolkit, environment ,version='latest', source='edm'):
+    """ Set up parameters dictionary for format() substitution """
+    edm_packages = test_dependencies | dependencies | toolkits.get(toolkit, set())
+
+    if source == 'edm':
+        pip_packages = set()
+        if version == 'latest':
+            edm_packages.add('enaml')
+        else:
+            edm_packages.add('enaml^={}'.format(version))
+    elif source == 'pypi':
+        pip_packages = pypi_dependencies
+        if version == 'latest':
+            pip_packages.add('enaml')
+        else:
+            pip_packages.add('enaml=={}'.format(version))
+    elif source == 'github':
+        pip_packages = repo_dependencies
+        pip_packages.add(
+            'git+https://github.com/nucleic/enaml.git#egg=enaml')
```

Comment:
Why is this not just in the `repo_dependencies` set? You're not using it anywhere else but here.